File: numerology_api/src/numerology_api/models/fortune.py
from typing import Any, Optional

from pydantic import BaseModel, Field


class FortuneRequest(BaseModel):
    bazi: dict[str, Any] = Field(..., description="八字排盘完整结果")
    question: Optional[str] = Field(default=None, description="求测者想重点了解的问题")


class DimensionAnalysis(BaseModel):
    label: str
    score: int = Field(ge=0, le=100)
    overall: str
    insights: list[str]
    suggestions: list[str]
    timing: Optional[str] = None


class FortuneResponse(BaseModel):
    summary: str
    dimensions: dict[str, DimensionAnalysis]
    remarks: str

File: numerology_web/src/api/fortune.ts
import client from './client'
import type { BaZiResponse } from './types'

export interface DimensionAnalysis {
  label: string
  score: number
  overall: string
  insights: string[]
  suggestions: string[]
  timing?: string | null
}

export interface FortuneResponse {
  summary: string
  dimensions: Record<string, DimensionAnalysis>
  remarks: string
}

export interface FortuneRequest {
  bazi: BaZiResponse
  question?: string
}

export const fortuneApi = {
  async analyze(payload: FortuneRequest): Promise<FortuneResponse> {
    const resp = await client.post<FortuneResponse>('/fortune/analyze', payload, {
      timeout: 120000,
    })
    return resp.data
  },
}
